rflx/expression.py

```diff
@@ -40,26 +40,33 @@ class ProofResult(Enum):
 
 
 class Proof:
+    LOGIC = "QF_NIA"
+
     def __init__(self, expr: "Expr", facts: Optional[Sequence["Expr"]] = None):
         self.__expr = expr
         self.__facts = facts or []
         self.__result = ProofResult.UNSAT
+        self.__unknown_reason = "unknown"
```

Comment:
Is there any case where the default value of `__unknown_reason` is used? If not, I think it would be better to use an empty string as default value to prevent the impression that it is relevant in some case.

Reply:
No, it's only used when set to an error message iff the result is `UNKNOWN`. I made it an optional `str` type and set it to `None` in ea507e4c127f31ec08e31eb5f56f823638c9d191.